db/dcrpg/insightapi.go

```diff
@@ -0,0 +1,209 @@
+package dcrpg
+
+import (
+	"errors"
+	"github.com/dcrdata/dcrdata/db/dbtypes"
+	apitypes "github.com/dcrdata/dcrdata/dcrdataapi"
+	"github.com/dcrdata/dcrdata/explorer"
+	"github.com/dcrdata/dcrdata/rpcutils"
+	"github.com/dcrdata/dcrdata/txhelpers"
+	"github.com/decred/dcrd/dcrjson"
+	"github.com/decred/dcrd/dcrutil"
+)
+
+// GetRawTransaction is used to fetch transaction verbose using
+// transaction id
+func (pgb *ChainDB) GetRawTransaction(txid string) (*dcrjson.TxRawResult, error) {
+	txraw, err := rpcutils.GetTransactionVerboseByID(pgb.client, txid)
+	if err != nil {
+		log.Errorf("GetRawTransactionVerbose failed for: %s", txid)
+		return nil, err
+	}
+	return txraw, nil
+}
+
+// GetBlockHeight returns a block height using the hash
+func (pgb *ChainDB) GetBlockHeight(hash string) (int64, error) {
+	height, err := RetrieveBlockHeight(pgb.db, hash)
+	if err != nil {
+		log.Errorf("Unable to get block height for hash %s: %v", hash, err)
+		return -1, err
+	}
+	return height, nil
+}
+
+// GetHeight returns the current best block height
+func (pgb *ChainDB) GetHeight() int {
+	height, _, _, _ := RetrieveBestBlockHeight(pgb.db)
+	return int(height)
+}
+
+// SendRawTransaction returns the transaction id
+// accepts the transaction hex
+func (db *ChainDB) SendRawTransaction(txhex string) (string, error) {
+	msg := txhelpers.MsgTxFromHex(txhex)
+	if msg == nil {
+		log.Errorf("SendRawTransaction failed: could not decode hex")
+		return "", errors.New("Could not decode hex")
+	}
+	hash, err := db.client.SendRawTransaction(msg, true)
+	if err != nil {
+		log.Errorf("SendRawTransaction failed: %v", err)
+		return "", err
+	}
+	return hash.String(), err
+}
+
+// InsightGetAddressTransactions returns transactions result
+func (pgb *ChainDB) InsightGetAddressTransactions(addr string, count, skip int) []*dcrjson.SearchRawTransactionsResult {
+	address, err := dcrutil.DecodeAddress(addr)
+	if err != nil {
+		log.Infof("Invalid address %s: %v", addr, err)
+		return nil
+	}
+	txs, err := pgb.client.SearchRawTransactionsVerbose(address, skip, count, true, true, nil)
+	if err != nil {
+		log.Warnf("GetAddressTransactions failed for address %s: %v", addr, err)
+		return nil
+	}
+	return txs
+}
+
+// GetTransactionHex returns hex representation of
+// a transaction
+func (pgb *ChainDB) GetTransactionHex(txid string) string {
+	txraw, err := rpcutils.GetTransactionVerboseByID(pgb.client, txid)
+
+	if err != nil {
+		log.Errorf("GetRawTransactionVerbose failed for: %v", err)
+		return ""
+	}
+
+	return txraw.Hex
+}
+
+// GetBlockVerboseByHash returns the verbose of a block by the block hash
```

Comment:
"the verbose of a block" doesnt' make much sense.  If you're not sure, just say it returns a dcrjson.GetBlockVerboseResult